### PR TITLE
Feat/dark mode

### LINKED ISSUE #76: Add dark mode/light theme - Vue
# Add dark mode/light theme with Tailwind CSS - Vue

## 🚀 Goal
Implement a dark mode/light theme toggle functionality in our Vue.js Decentralized Web App (DWA) starter using Tailwind CSS. Users should be able to switch between dark and light themes, and the app should remember their preference.

## 🤔 Background
We're expanding our DWA starter collection to include Vue.js. This task focuses on adding a theme toggle feature using Tailwind CSS, which is a common and user-friendly functionality in modern web applications.

> This is part of our larger project to create a Vue.js DWA starter. See our [main issue here](https://github.com/TBD54566975/tbd-examples/issues/63) for the full context and list of all related tasks.

## 🔑 Tasks and Acceptance Criteria
- [ ] Set up Tailwind CSS in the Vue.js project if not already done
- [ ] Configure Tailwind CSS for dark mode (using the `dark:` variant)
- [ ] Implement a reactive state to manage the current theme (dark or light)
- [ ] Create a toggle button or switch component using Tailwind classes for users to change between dark and light themes
- [ ] Ensure the theme change is applied immediately when toggled by adding/removing the `dark` class on the `<html>` tag
- [ ] Store the user's theme preference in local storage
- [ ] On app load, check local storage for the user's theme preference and apply it
- [ ] If no preference is stored, default to the user's system preference using Tailwind's `dark` mode setting
- [ ] Update all existing components and pages to use Tailwind's dark mode classes
- [ ] Add appropriate aria labels and roles to the theme toggle for accessibility
- [ ] Test the theme toggle functionality across different routes in the app

## 🌟 Resources
- [Tailwind CSS Dark Mode documentation](https://tailwindcss.com/docs/dark-mode)
- [Vue.js Documentation](https://vuejs.org/guide/introduction.html)
- [Tailwind CSS Documentation](https://tailwindcss.com/docs)
- [Window.localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)

## Getting Started
1. Comment ".take" on this issue to get assigned
2. Fork the repository and create a new branch for this task
3. Follow the tasks outlined above
4. Submit a pull request with your changes
5. Respond to any feedback during the review process

## Questions?
If you have any questions or need clarification, please comment on this issue or join our [Discord](https://github.com/TBD54566975/tbd-examples/issues/64#) community.

Happy coding! 🎉

### PR DESCRIPTION
**Pull request by Johnnyevans32**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 👷 Example Application
- [ ] 🧑‍💻 Code Snippet
- [ ] 🎨 Design
- [ ] 📖 Content
- [ ] 🧪 Tests
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This PR adds dark mode toggling.

## Related Tickets & Documents

Resolves #76 

## Mobile & Desktop Screenshots/Recordings

https://github.com/user-attachments/assets/82e9170a-a38b-4fed-9950-4c01b4e82d30

## Added code snippets?
- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 general documentation
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by Johnnyevans32** [comment]
@leordev @acekyd @blackgirlbytes @finn-tbd ready for review

File: javascript/dwa-starter-vue/tests/App.spec.ts [hunk 1]
```diff
@@ -22,64 +25,70 @@ const router = createRouter({
...
+    expect(document.documentElement.classList.contains('dark')).toBe(true)
+    expect(localStorage.getItem('vueuse-color-scheme')).toBe('dark')
+
+    // Set to auto mode and check
+    colorMode.value = 'auto'
+    await nextTick()
+
+    expect(localStorage.getItem('vueuse-color-scheme')).toBe('auto')
+  })
 })
```
Comment (blackgirlbytes) [inline javascript/dwa-starter-vue/tests/App.spec.ts:94]: awesome adding tests for dark mode and light mode!

**Review by blackgirlbytes — APPROVED** [review]
this is looking good to me! Just want to make sure that folks who are experts on vue like @acekyd and folks who are experts on DWAs like @leordev think the same. But I think this is ready to merge!

**Comment by blackgirlbytes** [comment]
Actually, I'm going to merge this to get some of the open PRs out of the way. It doesn't seem like there are any issues to me.

**Pull request merged**